### PR TITLE
HIP-584: Provide implementations for token metadata precompiles

### LINKED ISSUE #5599: HIP-584: Provide integration tests for token metadata precompiles
Provide integration tests for:

- GetTokenDefaultFreezeStatus
- GetTokenDefaultKycStatus
- GetTokenTypePrecompile
- IsFrozenPrecompile
- IsKycPrecompile

### PR DESCRIPTION
**Pull request by 0xivanov**
**Description**:
Provide precompile implementations for the following precompiles:

- GetTokenDefaultFreezeStatus
- GetTokenDefaultKycStatus
- GetTokenKeyPrecompile
- GetTokenTypePrecompile
- IsFrozenPrecompile
- IsKycPrecompile
- IsTokenPrecompile. 

**Related issue(s)**:

Fixes #5599

**Notes for reviewer**:
This change is Work In Progresss.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/Precompile.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        if (!Objects.equals(Wei.ZERO, frame.getValue())) {
+            final String INVALID_TRANSFER_MSG = "Transfer of Value to Hedera Precompile";
+            final Bytes INVALID_TRANSFER = Bytes.of(INVALID_TRANSFER_MSG.getBytes(StandardCharsets.UTF_8));
+            frame.setRevertReason(INVALID_TRANSFER);
+            frame.setState(REVERT);
+            throw new InvalidTransactionException(INVALID_FEE_SUBMITTED, "", "");
+        }
+    }
+
+    default Bytes getSuccessResultFor(final ExpirableTxnRecord.Builder childRecord) {
```
[the remark this answers is not in the document]
Comment (ivan-02) [reply hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/Precompile.java]: Done

File: hedera-mirror-web3/src/main/java/com/hedera/services/state/submerkle/ExpirableTxnRecord.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.services.state.submerkle;
+
+public interface ExpirableTxnRecord {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/state/submerkle/ExpirableTxnRecord.java:18]: We should delete this interface

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/impl/GetTokenTypePrecompile.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    @Override
+    public void body(final Bytes input, final UnaryOperator<byte[]> aliasResolver) {
+        final var tokenInfoWrapper = decodeGetTokenType(input);
+        tokenId = tokenInfoWrapper.token();
+    }
+
+    @Override
+    public Bytes getSuccessResultFor() {
+        //final var token = ledgers.getImmutableRef(tokenId);
+        //validateTrue(token != null, ResponseCodeEnum.INVALID_TOKEN_ID);
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/impl/GetTokenTypePrecompile.java:51]: We need this validation to check if the token exists

**Comment by steven-sheehy** [comment]
Can this PR be closed and the branch deleted?

**Pull request closed without merging**